File: lc167.py
from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
            lp = 0
            rp = len(numbers) - 1
            while lp < rp:
                sum = numbers[lp] + numbers[rp]
                if sum == target:
                     return [lp+1, rp+1]
                elif sum < target:
                    lp += 1
                elif sum > target:
                    rp -= 1

# Creating an instance of the Solution class
solution = Solution()

# # Calling the twoSum method on the instance
result = solution.twoSum([1, 2, 3], 5)
print(result)

File: two pointers/lc167.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {

    let lp = 0
    let rp = numbers.length-1
    let sum = numbers[lp] + numbers[rp]

    while (lp < rp && sum != target){
        if (sum < target){
            lp++
        } else if (sum > target){
            rp--
        }
        sum = numbers[lp] + numbers[rp]
    }

    return [lp+1, rp+1]
};